### PR TITLE
Rc/oauth

### PR DESCRIPTION
**Pull request by rctq**

### CODE REVIEW

File: tq42/client.py [hunk 1]
```diff
@@ -177,38 +214,51 @@ def login(self):
...
-                refresh_token = json_token["refresh_token"]
-                utils.save_token(
-                    service_name="refresh_token",
-                    backup_save_path=self.refresh_token_file_path,
-                    token=refresh_token,
-                )
-                current_datetime = datetime.now()
-                file_handling.write_to_file(self.timestamp_file_path, current_datetime)
+
+            if self.save_access_token(response_json) and self.save_refresh_token(
```
Comment (sbtq) [inline tq42/client.py:220]: executing a function that is called `save_` inside an if statement is an absolute no-go for me.

Please refactor this logic to actually indicate more what is happening.

File: tq42/client.py [hunk 2]
```diff
@@ -177,38 +214,51 @@ def login(self):
...
+                response_json
+            ):
                 break
 
             # Otherwise, wait for the specified interval before polling again
             time.sleep(interval)
 
+    def save_access_token(self, response: json):
+        if "access_token" in response:
```
Comment (sbtq) [inline tq42/client.py:229]: same as below regarding guard clauses

**Review by sbtq — CHANGES_REQUESTED** [review]
Made some changes mainly around readability, please take a look :)

File: tq42/client.py (hunk 1 above)
Comment (rctq) [reply tq42/client.py:220]: May you explain further? The logic that is used here is equivalent to the previous one. How do you prefer the refactor logic to flow?

File: tq42/utils/utils.py [hunk 3]
```diff
@@ -132,3 +132,7 @@ def get_token(service_name: str, backup_save_path: str) -> str:
 
     except (NoKeyringError, InitError):
         return file_handling.read_file(backup_save_path)
+
+
+def retrieve_specific_token(specific_token: str, response: dict) -> str:
```
Comment (sbtq) [inline tq42/utils/utils.py:137]: this function seems rather trivial and is not really necessary I would say 🙈

**Review by sbtq — CHANGES_REQUESTED** [review]

File: tq42/client.py [hunk 4]
```diff
@@ -135,9 +148,40 @@ def __init__(self, alt_config_file=None):
...
+            self.save_access_token(access_token)
+        else:
+            raise AuthenticationError(
+                "No access token can be retrieved from the response."
+            )
 
     @handle_generic_sdk_errors
     def login(self):
+        if self.credential_flow_client_id and self.credential_flow_client_secret:
+            print("Using Client Credential Flow.")
```
Comment (sbtq) [inline tq42/client.py:179]: we can omit this print, right? Was it only for debug?

**Review by sbtq — APPROVED** [review]

**Pull request merged**